### PR TITLE
Add strength indicator support in `PasswordTextField`

### PR DESCRIPTION
**Pull request by zurdus**

### CODE REVIEW

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt [hunk 1]
```diff
@@ -78,9 +98,74 @@ public fun PasswordTextField(
...
+            1f,
+            OrbitTheme.colors.success.normal,
+        )
+    }
+
+public sealed interface PasswordStrength {
+    public object Weak : PasswordStrength
+    public object Fair : PasswordStrength
+    public object Strong : PasswordStrength
+}
```
Comment (hrach) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt:167]: If I'm not mistaken our current scale (in the context of the progress bar) is not just a three-value scale. So it is a question of how to make it more scalable.

Currently, I'm instead thinking about this API:

```kotlin
(Password)TextField(
    validationContent = {
        PasswordStrength(
            value = 0.6,
            color = PasswordStrenth.Strong,
            label = { Text("Strong") },
        )
    }
)
```

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt [hunk 2]
```diff
@@ -78,9 +98,74 @@ public fun PasswordTextField(
...
         },
         interactionSource = interactionSource,
+        supportView = {
+            AnimatedVisibility(
+                visible = passwordStrength != null,
+                enter = slideInVertically() + fadeIn(),
+                exit = slideOutVertically() + fadeOut(),
+            ) {
+                if (passwordStrength != null) {
+                    Spacer(Modifier.size(4.dp))
```
Comment (gerak-cz) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt:108]: `.height` is enough

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt (hunk 1 above)
Comment (gerak-cz) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt:167]: That is true. There are multiple steps with the same color.

Check out the Change Password screen.

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt [hunk 3]
```diff
@@ -78,9 +98,74 @@ public fun PasswordTextField(
...
 
+@Composable
+private fun PasswordStrengthFooter(
+    passwordStrength: PasswordStrength,
+    strengthLabel: @Composable () -> Unit,
+) {
+    val (progress, color) = getPasswordStrengthValues(passwordStrength)
+    val progressAnimation by animateFloatAsState(progress)
+
+    Row(
```
Comment (gerak-cz) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt:126]: We will need to add some semantics here. We currently have UI tests that test the text on the strength indicator. That is just a one option. Please check out the Change Password screen.

File: catalog/src/main/java/kiwi/orbit/compose/catalog/screens/TextFieldScreen.kt [hunk 4]
```diff
@@ -104,11 +111,38 @@ private fun TextFieldScreenInner() {
...
+            validationContent = {
+                AnimatedVisibility(
+                    visible = passwordStrength != null,
+                    enter = slideInVertically() + fadeIn(),
+                    exit = slideOutVertically() + fadeOut(),
+                ) {
+                    if (passwordStrength == null) return@AnimatedVisibility
+
+                    PasswordStrengthIndicator(
+                        value = password.length / 10f,
```
Comment (hrach) [inline catalog/src/main/java/kiwi/orbit/compose/catalog/screens/TextFieldScreen.kt:137]: coerceAtMost(1f)

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordStrengthIndicator.kt [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            LocalContentColor provides color,
+            LocalTextStyle provides OrbitTheme.typography.bodySmallMedium,
+        ) {
+            label()
+        }
+    }
+}
+
+public enum class PasswordStrength {
+    Weak,
```
Comment (hrach) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordStrengthIndicator.kt:50]: Wouldn't be better to map them directly to color? Maybe also name it directly as PasswordStrengthColor?

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt [hunk 6]
```diff
@@ -89,6 +91,13 @@ internal fun PasswordTextFieldPreview() {
             value = "password",
             label = { Text("Password") },
             onValueChange = {},
+            validationContent = {
```
Comment (hrach) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordTextField.kt:94]: Originally I meant this slot named for both TextField and PasswordTextField, but if we are gonna use a different name, I guess we should name it strengthContent or similar, wdyt?

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/TextField.kt [hunk 7]
```diff
@@ -61,6 +63,7 @@ public fun TextField(
     enabled: Boolean = true,
     readOnly: Boolean = false,
     label: @Composable (() -> Unit)? = null,
+    supportContent: (@Composable () -> Unit)? = null,
```
Comment (hrach) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/TextField.kt:66]: I'm not fully convinced about this slot name. Maybe we could hide it from public API. What do you think?

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordStrengthIndicator.kt [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+) {
+    val progressAnimation by animateFloatAsState(value)
+
+    Row(
+        modifier = modifier.fillMaxWidth(),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically,
+    ) {
+        LinearProgressIndicator(
+            modifier = Modifier.width(163.dp),
```
Comment (Drekorian) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/PasswordStrengthIndicator.kt:35]: ?

**Comment by hrach** [comment]
Thanks for the all effort!

**Pull request merged**